### PR TITLE
Priya

### LINKED ISSUE #54: Display Window confirm and alert message in Delete Post

### LINKED ISSUE #56: Implement action and reducer for getfollowings  and hooked it up in UI
Implement action and reducer for getfollowings  and hooked it up in UI by passing username

### PR DESCRIPTION
**Pull request by slpriya**
Closed #54
Closed #56
Added deletePostById action in postActions.js
Added DELETE_POST types in types.js
Added DELETE_POST case in postReducers.js
show and hide thrash icon based on post postedBy
Implemented Front end logic for Delete Post  in PostActions.js
Added No Posts Yet in ProfilePosts if there is no posts.
Added window.confirm message before delete post.
Added Get followings Actions, reducers, and hooked it up in UI.
After uploading the avatar navbar avatar is not changing.
implemented action ,reducers of Following and UnfollowUser and hooked it up in UI.
Fixed error handling in login page (User not found error message ).
show Followers and Following component as overlay in ProfileHeader only if the user has followersCount and following count >0
Added follow user action reducer and linked it with search component in navbar
Added spinner in upload avatar, createposts, 
Added Clear_current_profile ,added clear_posts, actions and reducers called during logout user action in authActionsjs
Added get_followers actions and reducers hooked it up in UI
called logoutUser action during deleteAccount actions in profileActions

### CODE REVIEW

**Review by Emily484 — APPROVED** [review]
approved

File: client/src/components/Layout/Navbar.js [hunk 1]
```diff
@@ -16,32 +16,34 @@ import { logoutUser } from '../../actions/authActions';
...
 
+
   onLogoutClick(e) {
     e.preventDefault();
     this.props.logoutUser(this.props.history);
   }
 
   render() {
     const { isAuthenticated, user } = this.props.auth;
+    const { avatar } = this.props.profile;
```
Comment (solongoch) [inline client/src/components/Layout/Navbar.js]: this logic needs to be removed

File: client/src/components/Layout/Navbar.js [hunk 2]
```diff
@@ -113,7 +104,8 @@ class Navbar extends Component {
 }
 
 const mapStateToProps = (state) => ({
-  auth: state.auth
+  auth: state.auth,
+  profile: state.profile.profile
```
Comment (solongoch) [inline client/src/components/Layout/Navbar.js]: This line also needs to be removed

**Review by solongoch — APPROVED** [review]
Looks good. Keep comments

File: client/src/components/Layout/Navbar.js (hunk 2 above)
Comment (slpriya) [reply client/src/components/Layout/Navbar.js]: Done

**Pull request merged**